cornell-dti/office-hours
# Working version for discussions

### Summary 
Created the basic working version for the new discussions. 

I worked on the new version of discussions that includes lots of new features, which are listed below.

Here are the features that I worked on:
- Filter questions by if they are answered or unanswered
- Sort questions by upvote or most recent
- TAs/professors should be able to click the "Done" button to answer a question
- TAs/professors should be able to see the profile image and name of students who post questions, while normal students do not
- Answered questions should now have a resolved badge
- Updated the comment and upvote icons, and added new "sort by" arrows and resolved checkmark icons
- Normal students should not be able to comment on other student's questions
- No comments should be able to be added when discussion is over





### Test Plan 
- Create an unanswered question and log in as a TA/professor, clicking "Done" to see if a question gets sent from unanswered questions to answered questions (can be checked by changing the filter slider)
- Create multiple questions, some answered and some unanswered, and filter by if they are answered or unanswered to see if it works properly
- Try sorting multiple questions by upvote or most recent and see if it works (most upvotes or most recent should be at the top, depending on what you sort by)
- Log in as a student to make sure you can't see the names and profile pictures of people asking questions
- Try logging in as a TA/professor to make sure you can see the names and profile pictures of people who ask questions
- Check that any question in answered questions has the resolved badge
- Try commenting as a student on other students' questions and make sure this does not work
- Find a previous discussion that is already over or wait until your discussion you are using to test is over, and make sure you can not add comments
- Make sure all the new and updated icons look good


### Notes 



### Breaking Changes 

None
